### ISSUE TITLE
Sub/Sub components failing to pickup config

### ISSUE THREAD

**Issue by Guslington** [theonestack/cfhighlander#92]
This is my example

I have my base component which is cloud front
https://github.com/theonestack/hl-component-cloudfront

I then have another component that includes the cloudfront component and renders inline.
This is a s3 bucket and bucket policy
https://github.com/theonestack/hl-component-s3-cloudfront

i then have my project that specifies the s3-cloudfront component called s3-website

```ruby
CfhighlanderTemplate do
  Name 's3-website'
  Description "s3-website"

  Component name: 's3cloudfront', template: 's3-cloudfront', render: Inline

end
```
I then want add config to the cloudfront component, so i create a `cloudfront.config.yaml` with my cloudfront config. However when i compile the config is not picked up. The compiled cloudfront config only contains config set by the `cloudfront` and `s3-cloudfront` components

I have tried specifying 
```yaml
component:
  cloudfront:
```

proving the config in a 's3-cloudfront.config.yaml' file

The only way i can get this to work is by vendoring the `s3-cloudfront` component in my `s3-website` project and proving the cloudfront config with that vendored component.

@toshke @aaronwalker any ideas why it might not be applying the config?

**Comment by toshke** [comment]
@Guslington

cloudfront component is 3 levels deep and as such it won't get picked up by
highlander config file `cloudfront.config.yaml`

As for the other option - explicitly configuring cloudfront component
within s3cloufront config file, this should work. I'll play a bit with this
use case - perhpas having `component.subcomponent.config.yaml`
as auto loaded config file for auch structure.

**Comment by toshke** [comment]
Implemented in #97
